internal/abci/abci.go

```diff
@@ -197,6 +202,11 @@ func (a *AbciApp) CheckTx(ctx context.Context, incoming *abciTypes.RequestCheckT
 		return &abciTypes.ResponseCheckTx{Code: code.Uint32(), Log: err.Error()}, nil
 	}
 
+	// Cache the transaction hash
+	if newTx {
+		txHash := cmtTypes.Tx(incoming.Tx).Hash()
+		a.txCache[string(txHash)] = true
```

Comment:
We have to worry about data races here.  `TxInMempool` can be called on demand via RPC, correct?

Reply:
Oh shoot, totally forgot about those. Thanks for bringing that up

Reply:
steer away from `sync.Map` tho

Reply:
oh nice you did it already with rwmutex :partying_face: